src/test/java/net/snowflake/client/jdbc/DatabaseMetaDataLatestIT.java

```diff
@@ -206,6 +206,70 @@ public void testDoubleQuotedDatabaseInGetSchemas() throws SQLException {
     }
   }
 
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testDoubleQuotedDatabaseInGetTables() throws SQLException {
+    try (Connection con = getConnection()) {
+      Statement statement = con.createStatement();
+      // Create a database with double quotes inside the database name
+      statement.execute("create or replace database \"dbwith\"\"quotes\"");
+      statement.execute("create or replace schema \"dbwith\"\"quotes\".\"testschema\"");
+      statement.execute(
+          "create or replace table \"dbwith\"\"quotes\".\"testschema\".\"testtable\" (col1 string, col2 string)");
+      DatabaseMetaData metaData = con.getMetaData();
+      ResultSet rs = metaData.getTables("dbwith\"quotes", "testschema", null, null);
+      assertEquals(1, getSizeOfResultSet(rs));
+      rs.close();
+      statement.close();
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
+  public void testDoubleQuotedDatabaseInGetColumns() throws SQLException {
+    try (Connection con = getConnection()) {
+      Statement statement = con.createStatement();
+      // Create a database with double quotes inside the database name
+      statement.execute("create or replace database \"dbwith\"\"quotes\"");
+      statement.execute("create or replace schema \"dbwith\"\"quotes\".\"testschema\"");
+      statement.execute(
+          "create or replace table \"dbwith\"\"quotes\".\"testschema\".\"testtable\"  (col1 string, col2 string)");
+      DatabaseMetaData metaData = con.getMetaData();
+      ResultSet rs = metaData.getColumns("dbwith\"quotes", "testschema", null, null);
+      assertEquals(2, getSizeOfResultSet(rs));
+      rs.close();
+      statement.close();
+    }
+  }
+
+  @Test
+  @ConditionalIgnoreRule.ConditionalIgnore(condition = RunningOnGithubAction.class)
```

Comment:
The tests look good! I just realized there are a few other functions where this same issue may pop up. Could you check if this is also an issue for:

getFunctions()
getFunctionColumns()
getProcedures()
getProcedureColumns()
getTablePrivileges()

Reply:
For getFunctionColumns() and getProcedureColumns() it's more complicated because both of these functions have a two step process where the functions/procedures are retrieved first and then a second query is used to get the columns for each. So getProcedureColumns first calls getProcedures for the given database and returns a result set. https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java#L1109
Then for getProcedureColumns it uses the result set from the previous call to get the schema name for the procedure https://github.com/snowflakedb/snowflake-jdbc/blob/master/src/main/java/net/snowflake/client/jdbc/SnowflakeDatabaseMetaData.java#L1139
but since Snowflake returns database and schema names that have special characters already enclosed in double quotes, if we were to try escape double quotes in the name it would also try to escape the quotes that are enclosing the name and would return empty results

Reply:
oh true, maybe it wouldn't work for those.